### DISCUSSION TITLE
Fully-managed react-email, inside Resend

### DISCUSSION THREAD

**Discussion by thiskevinwang** [resend/react-email#976] [Ideas]
### Goals

1. As a designer and/or engineer I want to be able to iterate on a template, via some form of `react-email`, directly inside Resend
   - templates should have revision history, drafts, and be branchable — think planetscale/neon db branching
   - templates should have an ID
   - templates should expect a JSON props interface
2. As an engineer, I want to be able to send an email by simply passing an `apikey`, `template_ID`, and a JSON blob that satisfies the interface from above, to Resend.

### Background

At Clerk, we have a `go` backend which is also where a lot of SQL queries happen, for customer data. This customer data is eventually injected into `.txt` email templates (today), and maybe soon,`react-email`-outputted HTML templates.
- using Resend's simple `go` SDK here is a great DX

In a separate playground repo, i'm using `react-email` and setting various strings in rendered HTML like `{{thing}}` so that resulting output can be used as a template in our `go` backend's template engine of choice (ie. handlebars). 
- this awkward step of outputting, HTML, and pasting it into the `go` codebase is an awkward DX

It would be nice to eliminate this separate repo, and lean into Resend as platform more.  😍

### Proposal

This should exist as a fully-managed feature.

**Comment by gabrielmfern** [comment]
I like the ideas a lot! 
When you say having the templates on Resend, do you mean the actual HTML or the actual component code for like `react-email`?

Also, would you find it easier to iterate on the emails from Resend, or from your local codebase? I ask that because a nice idea that maybe might solve this for you, would be to have some sort of SDK for each kind of back-end language that helps render the emails even with passing props into the components. A `@react-email/render` for most back-end languages I mean.

**Comment by shahriyardx** [comment]
Like codepen where we can write the code and see the preview, then send it to someone? That would be great, when I heared about resend, I thought they might have this, but actually not

**Comment by gabrielmfern** [comment]
Bit of an update to this, Zeno and Bu talked about something like this on their [Q&A](https://www.youtube.com/watch?v=WN5RQGmBB-A&t=522s)
